.github/workflows/test_macos.yaml

```
@@ -0,0 +1,87 @@
+name: macOS Test
+
+on:
+  push:
+    paths:
+      - 'osx/**'
+      - '.github/workflows/test_macos.yaml'
+  pull_request:
+    paths:
+      - 'osx/**'
+      - '.github/workflows/test_macos.yaml'
+
+jobs:
+  unit_test:
+    defaults:
+      run:
+        shell: ${{ matrix.run_wrapper || 'bash --noprofile --norc -eo pipefail {0}' }}
+    runs-on: ${{ matrix.runs_on || 'macos-latest' }}
+    strategy:
+      matrix:
+        include:
+          - runs_on: macos-latest
+            python: 3.9
+          - runs_on: apple-silicon-m1
+            run_wrapper: arch -arm64 bash --noprofile --norc -eo pipefail {0}
+            python: 3.9.7
```

```suggestion
            python: 3.9.7
    name: macOS Test ( ${{ matrix.runs_on }} )
```

Just so that it's easier to read.